Add annotations for `chrome.notifications`

ssorallen opened the pull request:
[chrome.notifications][1]

* `chrome.notifications.create` has a signature best described using
  unions for each of its parameters rather than a union of multiple
  full function signatures.

[1]: https://developer.chrome.com/apps/notifications

ssorallen committed:
Add annotations for `chrome.notifications`

[chrome.notifications][1]

* `chrome.notifications.create` has a signature best described using
  unions for each of its parameters rather than a union of multiple
  full function signatures.

[1]: https://developer.chrome.com/apps/notifications

ssorallen commented:
Flow currently does not support [function overloading in object types](https://github.com/facebook/flow/issues/1556), so `chrome.create` cannot easily be written with its 3 possible signatures. While the version in this PR is not perfectly correct (it would be valid to call it with `create(chrome$NotificationOptions, chrome$NotificationOptions)`, for example), the version here is helpful.

Shraymonks commented on interfaces/notifications.js:
```diff
@@ -0,0 +1,58 @@
+
+type chrome$notifications = {
+  clear(notificationId: string, callback?: (wasCleared: boolean) => void): void,
+  create(
```
You can do something like this instead as a workaround:
```
create: (
  ((
    notificationId: string,
    options: chrome$NotificationOptions,
    callback?: (notificationId: string) => void
  ) => void) &
  ((
    options: chrome$NotificationOptions,
    callback?: (notificationId: string) => void
  ) => void)
),
```

Shraymonks reviewed (changes requested):
Thanks for the PR!

I think the real issue here is that flow doesn't currently support non-trailing optional parameters.

ssorallen committed:
Use union to support exact notification.create signatures

The first parameter, `notificationId`, is now effectively optional as is
supported by the API.

ssorallen commented:
@Shraymonks I added your suggested change. It works fine. Let me know what you think.

Shraymonks commented:
Thanks!

ssorallen commented:
@Shraymonks Could you release a version of the NPM package with this PR in it? I am using this package in a project and needed the notification types!

Shraymonks commented:
@ssorallen Done!
